src/cpp/core/system/PosixChildProcess.cpp

```diff
@@ -906,6 +927,7 @@ void AsyncChildProcess::poll()
    pAsyncImpl_->initTimers(options().reportHasSubprocs);
    if (pAsyncImpl_->checkChildCount())
    {
+      pAsyncImpl_->hasRecentOutput_ = false;
```

Comment:
If there is a long-running async child process that ISN'T a terminal (and thus doesn't do child process counting), then both these flags would stay "true", thus seen as busy and would prevent suspending.